### PR TITLE
Removes Krav Maga Gloves from wardens locker and instead makes warden start with a skill chip

### PR DESCRIPTION
**Pull request by SomeRandomOwl**
## About The Pull Request

Try number two as my previous PR for this was a little dirty and had remenants of a previous PR i made that was already merged

This PR modifies `code\game\objects\structures\crates_lockers\closets\secure\security.dm` to and removes `new /obj/item/clothing/gloves/krav_maga/sec(src) ` and replaces it with `/obj/item/skillchip/job/warden` which is a new skill chip that warden will start with in their loadout

## How This Contributes To The Skyrat Roleplay Experience

It replaces a relatively powerful item that can be easily stolen off of the warden with something not so easily stolen, and also from what I recall there was talk about doing this replacement, but it was never done, so I am simply doing it now with this PR. 

## Proof of Testing

<details>
<summary>Screenshots/Videos

image
image
image
image

</summary>
  
</details>

## Changelog

:cl:
add: New skill chip for Krav Maga and adds it to the wardens loadout
balance: Removes the Krav Maga gloves from the wardens locker

### CODE REVIEW

**Comment by Paxilmaniac** [comment]
image

**Comment by SomeRandomOwl** [comment]
> image

Why would that be the case? This PR is specifically made for SR as it was previously requested by the host

**Review by tf-4 — COMMENTED** [review]
It's a good thing to be able to steal the gloves.
Closing in 24h.

**Comment by PCPort** [comment]
Isn't the gloves being able to be stolen kind of the point? If you don't want to lose them, quit getting into fights that you can't win as a warden.

**Comment by SomeRandomOwl** [comment]
Honetly made this because I remember it was brought up a while ago and i talked about it again recently and someone mentioned it, i thought id try to do this PR but I didnt relly know if it would be accepted or not as it is kinda contraversal at best, though it could probally be done in a different method, but for now ill just manually close this myself

**Comment by hugmeorelse** [comment]
I asked for this.

**Comment by SomeRandomOwl** [comment]
i could change this to make the Warden start with the implant instead of having a implanter in the locker

**Review by Zonespace27 — CHANGES_REQUESTED** [review]
If you want to solve the problem that this was meant to solve, make it a skillchip auto-implanted in the warden by default

**Comment by hugmeorelse** [comment]
> If you want to solve the problem that this was meant to solve, make it a skillchip auto-implanted in the warden by default

Is this something you can do possibly instead? @SomeRandomOwl

**Comment by SomeRandomOwl** [comment]
> > If you want to solve the problem that this was meant to solve, make it a skillchip auto-implanted in the warden by default
> 
> Is this something you can do possibly instead? @SomeRandomOwl

looking into it right now

File: modular_skyrat/modules/skillchips/warden.dm [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	name = "Krav Maga skillchip"
+	desc = "Teaches you the arts of Krav Maga in 5 short instructional videos beamed directly into your eyeballs."
+	skill_name = "Krav Maga"
+	skill_description = "A specialised form of Martial Arts, allowing you to swiftly disable attackers."
+	skill_icon = "brain"
+	activate_message = "<span class='notice'>You can visualize how to defend yourself with martial arts.</span>"
+	deactivate_message = "<span class='notice'>You forget how to control your muscles to execute the arts of Krav Maga.</span>"
+	var/datum/action/Krav_Activate/Krav_Activate = new/datum/action/Krav_Activate()
+
+/datum/action/Krav_Activate
```
Comment (Zonespace27) [inline modular_skyrat/modules/skillchips/warden.dm]: don't capitalize paths

File: modular_skyrat/modules/skillchips/warden.dm [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	name = "Activates your Knowledge of Krav Maga"
```
Comment (Zonespace27) [inline modular_skyrat/modules/skillchips/warden.dm]: add a period, also knowledge isn't a proper noun

File: modular_skyrat/modules/skillchips/warden.dm [hunk 3]
```diff
@@ -0,0 +1,41 @@
+/obj/item/skillchip/job/warden
+	name = "Krav Maga skillchip"
+	desc = "Teaches you the arts of Krav Maga in 5 short instructional videos beamed directly into your eyeballs."
+	skill_name = "Krav Maga"
+	skill_description = "A specialised form of Martial Arts, allowing you to swiftly disable attackers."
+	skill_icon = "brain"
+	activate_message = "<span class='notice'>You can visualize how to defend yourself with martial arts.</span>"
+	deactivate_message = "<span class='notice'>You forget how to control your muscles to execute the arts of Krav Maga.</span>"
+	var/datum/action/Krav_Activate/Krav_Activate = new/datum/action/Krav_Activate()
```
Comment (Zonespace27) [inline modular_skyrat/modules/skillchips/warden.dm]: remove this on Destroy(), also path after new is implicit

File: modular_skyrat/modules/skillchips/warden.dm [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+/datum/action/Krav_Activate
+	name = "Activates your Knowledge of Krav Maga"
+	button_icon = 'icons/obj/wizard.dmi'
+	button_icon_state = "scroll2"
+
+/datum/action/Krav_Activate/Trigger(trigger_flags)
+	var/mob/living/carbon/human/H = owner
```
Comment (Zonespace27) [inline modular_skyrat/modules/skillchips/warden.dm]: one letter vars bad

File: modular_skyrat/modules/skillchips/warden.dm [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+	var/datum/martial_art/krav_maga/style = new
```
Comment (Zonespace27) [inline modular_skyrat/modules/skillchips/warden.dm]: if they already have it, don't make a `new` datum that has to be deleted post-proc

File: modular_skyrat/modules/skillchips/warden.dm [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+	else
+		style.teach(H, TRUE)
+
+/obj/item/skillchip/job/warden/on_activate(mob/living/carbon/user, silent = FALSE)
+	. = ..()
+	Krav_Activate.Grant(user)
+
+/obj/item/skillchip/job/warden/on_deactivate(mob/living/carbon/user, silent = FALSE)
+	. = ..()
+	Krav_Activate.Trigger()
```
Comment (Zonespace27) [inline modular_skyrat/modules/skillchips/warden.dm]: this doesn't account for them having it off

File: modular_skyrat/modules/skillchips/warden.dm [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+	Krav_Activate.Remove(user)
+	return ..()
+
+
```
Comment (Zonespace27) [inline modular_skyrat/modules/skillchips/warden.dm]: don't need all this whitespace after

**Review by Zonespace27 — CHANGES_REQUESTED** [review]
overall, I *heavily* recommend looking at /obj/item/skillchip/job/chef and just copying how it grants/removes a martial art instead of what you have currently

File: modular_skyrat/modules/skillchips/warden.dm (hunk 3 above)
Comment (SomeRandomOwl) [reply modular_skyrat/modules/skillchips/warden.dm]: I swapped it to just new, but im not certain what you mean to remove it on destroy

File: modular_skyrat/modules/skillchips/warden.dm (hunk 5 above)
Comment (SomeRandomOwl) [reply modular_skyrat/modules/skillchips/warden.dm]: I am not fully certain how to fix this

**Comment by SomeRandomOwl** [comment]
@Zonespace27 Do you want the skillchip to be limited to just security Areas? thats how the chef skillchip works, this simply makes it work like the Krav maga implant

File: modular_skyrat/modules/skillchips/warden.dm (hunk 5 above)
Comment (Zonespace27) [reply modular_skyrat/modules/skillchips/warden.dm]: move this into the `else` clause on line 25-26

File: modular_skyrat/modules/skillchips/warden.dm [hunk 8]
```diff
@@ -0,0 +1,37 @@
+/obj/item/skillchip/job/warden
+	name = "Krav Maga skillchip"
+	desc = "Teaches you the arts of Krav Maga in 5 short instructional videos beamed directly into your eyeballs."
+	skill_name = "Krav Maga"
+	skill_description = "A specialised form of Martial Arts, allowing you to swiftly disable attackers."
+	skill_icon = "brain"
+	activate_message = "<span class='notice'>You can visualize how to defend yourself with martial arts.</span>"
+	deactivate_message = "<span class='notice'>You forget how to control your muscles to execute the arts of Krav Maga.</span>"
+	var/datum/action/krav_activate/krav_activate = new
```
Comment (Zonespace27) [inline modular_skyrat/modules/skillchips/warden.dm]: on `/obj/item/skillchip/job/warden/Destroy()`, QDEL_NULL this

File: code/modules/jobs/job_types/warden.dm [hunk 9]
```diff
@@ -69,3 +69,4 @@
 
 	box = /obj/item/storage/box/survival/security
 	implants = list(/obj/item/implant/mindshield)
+	skillchips = list(/obj/item/skillchip/job/warden) //SKYRAT EDIT ADDITION
```
Comment (Zonespace27) [inline code/modules/jobs/job_types/warden.dm]: modularize this

File: modular_skyrat/modules/skillchips/warden.dm [hunk 10]
```diff
@@ -0,0 +1,37 @@
+/obj/item/skillchip/job/warden
+	name = "Krav Maga skillchip"
```
Comment (Zonespace27) [inline modular_skyrat/modules/skillchips/warden.dm]:
```suggestion
	name = "\improper Krav Maga skillchip"
```

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

File: modular_skyrat/modules/skillchips/warden.dm (hunk 10 above)
Comment (SomeRandomOwl) [reply modular_skyrat/modules/skillchips/warden.dm]: I cam confused with this one? What change is needed here @Zonespace27

**Comment by Iamgoofball** [comment]
:-1: to putting it in the warden's head roundstart, needs to be in the locker or orderable via cargo somehow so that it's usable on a job changed warden, i added krav maga back in the day

making it a skill chip is fine i guess but i'd deactivate it if they've got chunky fingers, otherwise this is a massive buff to krav maga and a reduction in ability to job transfer people to positions and have the gear be available
1. krav maga being a skillchip means you can't disarm it off of someone by literally disarming them or by forcefully removing their gloves, you gotta surgery it out now
2. krav maga being a skillchip means you can stack it with insulated gloves now making it significantly more powerful than intended in combat situations, heavily advise making the krav maga moves not work if you've got chunky fingers to accomodate

File: modular_skyrat/modules/skillchips/warden.dm (hunk 10 above)
Comment (Zonespace27) [reply modular_skyrat/modules/skillchips/warden.dm]: add the `\improper` label to the name as a prefix

**Comment by SomeRandomOwl** [comment]
> 👎 to putting it in the warden's head roundstart, needs to be in the locker or orderable via cargo somehow so that it's usable on a job changed warden, i added krav maga back in the day
> 
> making it a skill chip is fine i guess but i'd deactivate it if they've got chunky fingers, otherwise this is a massive buff to krav maga and a reduction in ability to job transfer people to positions and have the gear be available
> 
>     1. krav maga being a skillchip means you can't disarm it off of someone by literally disarming them or by forcefully removing their gloves, you gotta surgery it out now
> 
>     2. krav maga being a skillchip means you can stack it with insulated gloves now making it significantly more powerful than intended in combat situations, heavily advise making the krav maga moves not work if you've got chunky fingers to accomodate

I do agree with you on point 2, im looking at ways to prevent them using insulated gloves with the skillchip but this was specifically requested by the host to be in the warden outfit loadout instead of the locker that is the primary reason this PR was created

**Comment by Iamgoofball** [comment]
have you tried asking for a reason before blindly doing things? you can easily miss potentially better solutions to a problem as a result if you never question an approach to something before implementing it

**Comment by Zergspower** [comment]
That's a far reach in assumptions as sec getting insuls is something we rarely if ever see here. Also check your tone, you're being a complete ass for no reason.

> have you tried asking for a reason before blindly doing things? you can easily miss potentially better solutions to a problem as a result if you never question an approach to something before implementing it

**Comment by CliffracerX** [comment]
> have you tried asking for a reason before blindly doing things? you can easily miss potentially better solutions to a problem as a result if you never question an approach to something before implementing it

Discussion around this PR and the rebalance of Krav Maga has been ongoing in the community for a good couple days now.  If you were actually a part of that community, you'd know this wasn't just "blindly doing things", and you'd have had even less of an excuse to come be a jerk to someone still learning the ropes of contributing.

**Comment by softcerv** [comment]
> @Zonespace27 Do you want the skillchip to be limited to just security Areas? thats how the chef skillchip works, this simply makes it work like the Krav maga implant

I think this is a good idea. Otherwise it becomes a martial art that can be used anywhere on the station and can't easily be disarmed.

**Comment by Iamgoofball** [comment]
i am not "being a jerk" by saying that it's okay to question the words "host request", especially given the fact that "How This Contributes To The Skyrat Roleplay Experience" does not match the actual reason for the request and that the fact it was a host request was only pulled out upon criticism of the PR, which per two separate maintainers the actual reason for this PR is "the HoS keeps stealing it from the warden" which goes completely unmentioned in this PR, the PR description, and xyel's posts here

i am unsure as to why nobody wants to discuss that aspect of this, but given that it is the actual reason the PR is being made per two separate maintainers, i feel it's relevant to discuss and thus relevant to consider to the PR. 

xyel is **not** a child who's going to melt down at you if you question his opinions, dancing around his reasoning for the request simply makes it look like you think he's a baby who can't take criticism and you and I both know that's not only completely untrue, but it's also far, far more rude than me being blunt about the situation at hand here

it is not recommended to make big sweeping balance changes to things without documenting the reasoning for precisely this reasoning, important context behind a decision can easily get lost to those not privy to the internal discussions as a result, people should not have to be asking maintainers why a PR exists because the PR itself is devoid of any and all context and reasoning

xyel is correct that there's an issue if heads of security are stealing their subordinate's gear, but the solution to that is not to apply a massive buff to said gear, it is to change the conditions under which they're acquiring gear in the first place to the accepted standards for gear acquisition, such as cargo, thereby resolving this in a far more subtle way that lets admins do their job

overall, this PR is a massive, massive buff to krav maga and wardens by making them able to krav maga with insulated gloves, and unable to be disarmed of krav maga without surgery to remove their skill chip, to resolve a problem that isn't documented at all in the actual PR description, the comments before this, or the commit messages and comments in-code, this is a sledgehammer fix when we need a screwdriver fix

the codebase is not capable of nuance in situations where player behavior is subpar, we have a sledgehammer, the admins are the ones with the more subtle and nuanced tools to resolve this, so it should be up to them, we can only guide players towards good interactions with eachother, admins are the ones who enforce those good interactions if they ignore our guiding stones

**Comment by ChungusGamer666** [comment]
the host needs to make better requests

**Comment by SomeRandomOwl** [comment]
> > @Zonespace27 Do you want the skillchip to be limited to just security Areas? thats how the chef skillchip works, this simply makes it work like the Krav maga implant
> 
> I think this is a good idea. Otherwise it becomes a martial art that can be used anywhere on the station and can't easily be disarmed.

I can also look into doing this as this technically would nerf it for wardens who like to walk out of brig when there is action going on

**Comment by Iamgoofball** [comment]
it doesn't need a nerf if you aren't giving it a massive, unneeded buff in the first place, please justify your massive buff to krav maga and why it will improve the skyrat roleplaying experience because it's not mentioned at all in your PR description still

**Comment by YakumoChen** [comment]
> That's a far reach in assumptions as sec getting insuls is something we rarely if ever see here. 

That's because literally none of security's weapons work with insuls
This provides a means to enable exactly that.

No wonder goof made the other PR

**Comment by hugmeorelse** [comment]
Please keep comments pertinent to the PR. Thanks.

**Comment by hugmeorelse** [comment]
First of all, I think it's clear some context is needed.

**@SomeRandomOwl worked on this PR, because people were talking about issues with Krav Maga gloves as @CliffracerX points out.** During that discussion, someone brought up that I requested this change to @Zonespace27 a while back, and it gave some people some ideas on how to go about it. It's not something that was done unquestioningly. Also, I wasn't even aware at the time, since I'm currently in the middle of moving. A staff member DMed me about it, so I got involved to back them up.

Also, people are commenting on this without understand the problem, namely the fact that there is often conflict in regards to Head of Security stealing Krav Maga gloves from the warden frequently to the point it has become problematic. Call it powergaming if you like. It doesn't matter. Anyways, I understand the frustration with the lack of initial reasoning given, but that's because I was only made aware later and could only sporadically comment on it due to me moving.

Anyways, back to the topic on hand, it's a completely unnecessary problem that keeps rearing its ugly head, and I don't want this to be a ban bait for players. Whether or not the warden has krav maga, contrary what some people here assert, does not add much to the overall gameplay loop. I have rarely ever seen it ever become a focal point in antagonist gameplay. My main concern right now is putting a stop to issues that staff simply should not have to deal with. It is ultimately a minor change.

I also want to make one thing clear. I honestly could not care less if this PR was merged or closed. I forgot about it a long time ago, as it is, again, a minor change. However, this person is doing a PR that I requested to someone else a long time ago. It would be simply shitty of me to leave them out to dry, just because the PR is unpopular. They did not come up with the idea. I did.

tl;dr krav maga gloves are ban-bait. I want to stop it from being a headache for players and staff. People had a discussion about krav maga gloves, someone brought up I requested Zonespace to do something about it, and someone picked up on it. I was made aware later, and I got involved later, hence why the lack of initial reasoning why this is being done. Still busy with moving though.

**Comment by hugmeorelse** [comment]
Anyways, I mentioned this to the maintainers but if there are any alternative solutions out there, feel free to share. Either ways, I'll be talking to maintainers and staff about this. I hope to see new comments in regards to why this change was requested with this new context in mind. However, please refrain from being a dick. This person spent the time to fulfill a request I made a long time ago, and they don't deserve to be dog-piled on for that. As someone said, they are a new contributor. Be nice. It's not their fault anyways that this PR exists.  

Oh, also, again, I'm currently busy moving. So, responses may be sparse.

Also, I talked to Goofball. He has an alternative PR setup to address the problem I cited above. So, let's see how that is first. We can put this one on DNM for now.

_By the way, I'm editing this comment and the last one for further context, since much of the discussion I'm having is over Discord. I'm updating here as needed for full context for everyone reading. This is my last edit of the night before I get ready for bed._
